functions/snippets/src/main/java/com/example/functions/ImageMagick.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions;
+
+// [START functions_imagemagick_setup]
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Feature.Type;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageSource;
+import com.google.cloud.vision.v1.SafeSearchAnnotation;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class ImageMagick implements BackgroundFunction<GcsEvent> {
+
+  private static Storage storage = StorageOptions.getDefaultInstance().getService();
+  private static final String BLURRED_BUCKET_NAME = System.getenv("BLURRED_BUCKET_NAME");
+  private static final Logger LOGGER = Logger.getLogger(ImageMagick.class.getName());
+  // [END functions_imagemagick_setup]
+
+  // [START functions_imagemagick_analyze]
+  @Override
+  // Blurs uploaded images that are flagged as Adult or Violence.
+  public void accept(GcsEvent gcsEvent, Context context) {
+    BlobInfo blobInfo = BlobInfo.newBuilder(gcsEvent.getBucket(), gcsEvent.getName()).build();
+
+    // Construct URI to GCS bucket and file.
+    String gcsPath = String.format("gs://%s/%s", gcsEvent.getBucket(), gcsEvent.getName());
```

Comment:
We should make sure you have an empty bucket for this if you are letting users provide the event.

Reply:
There should be a `WARNING: In production, you would not want to allow users the ability to provide unsanitized requests directly to APIs.`

Reply:
This is a _storage_-triggered function - so I'm assuming the request can _only_ come from GCS?

(I assume it's still worth adding a comment to that effect, or maybe something like `WARNING: even though this request comes directly from Cloud Storage, it's still a good idea to sanitize it in production to prevent directory traversal attacks`)

Reply:
yes